graylog2-server/src/main/java/org/graylog2/shared/metrics/MetricUtils.java

```diff
@@ -72,38 +73,41 @@
     }
 
     public static Map<String, Object> map(String metricName, Metric metric) {
-        String type = metric.getClass().getSimpleName().toLowerCase(Locale.ENGLISH);
-
-        if (type.isEmpty()) {
-            type = "gauge";
-        }
-
-        Map<String, Object> metricMap = Maps.newHashMap();
+        final Map<String, Object> metricMap = Maps.newHashMap();
         metricMap.put("full_name", metricName);
         metricMap.put("name", metricName.substring(metricName.lastIndexOf(".") + 1));
-        metricMap.put("type", type);
 
         if (metric instanceof Timer) {
             metricMap.put("metric", buildTimerMap((Timer) metric));
+            metricMap.put("type", "timer");
         } else if(metric instanceof Meter) {
             metricMap.put("metric", buildMeterMap((Meter) metric));
+            metricMap.put("type", "meter");
         } else if(metric instanceof Histogram) {
             metricMap.put("metric", buildHistogramMap((Histogram) metric));
+            metricMap.put("type", "histogram");
+        } else if(metric instanceof Counter) {
+            metricMap.put("metric", ((Counter) metric).getCount());
+            metricMap.put("type", "counter");
+        } else if(metric instanceof Gauge) {
+            metricMap.put("metric", ((Gauge) metric).getValue());
+            metricMap.put("type", "gauge");
         } else {
             metricMap.put("metric", metric);
+            // best guess
+            metricMap.put("type", "gauge");
```

Comment:
Not sure if we should keep this "catch all" or simply return an error. 🤔

Reply:
What other types are there? If there aren't any other metric types, I think we should throw an error.

Reply:
AFAIK there aren't any others used by Graylog.

The Graylog web interface also only has support for Counter, Gauge, Histogram, Meter, and Timer.

https://github.com/Graylog2/graylog2-server/blob/0f97411137b2f5b36f3a24d0a74c409ef469df39/graylog2-web-interface/src/components/metrics/MetricDetails.jsx#L27-L36